File: app/src/androidTest/kotlin/br/com/alma/meustrocados/persistencia/TipoContaDAOTest.java
package br.com.alma.meustrocados.persistencia;

import org.junit.Assert;
import org.junit.Test;

import java.util.List;

import br.com.alma.meustrocados.modelo.TipoConta;

public class TipoContaDAOTest extends AppDatabaseTest {

    @Test
    public void insertTipoConta() throws Exception {
        TipoConta tipoConta = new TipoConta(1, "TipoConta");
        TipoConta returningValue;

        getTipoContaDAO().insert(tipoConta);
        returningValue = getTipoContaDAO().getTipoConta(tipoConta.uid);

        Assert.assertEquals(tipoConta.uid, returningValue.uid);
    }

    @Test
    public void deletetipoConta() throws Exception {
        TipoConta tipoConta = new TipoConta(1, "TipoConta");
        TipoConta returningValue;

        getTipoContaDAO().insert(tipoConta);
        getTipoContaDAO().delete(tipoConta);
        returningValue = getTipoContaDAO().getTipoConta(tipoConta.uid);

        Assert.assertNull(returningValue);
    }


    @Test
    public void recuperaTodosOsTipoContas() throws Exception {
        TipoConta tipoConta = new TipoConta(1, "Banco");
        getTipoContaDAO().insert(tipoConta);
        tipoConta = new TipoConta(2, "Caixa");
        getTipoContaDAO().insert(tipoConta);
        tipoConta = new TipoConta(3, "Cartão");
        getTipoContaDAO().insert(tipoConta);

        List<TipoConta> returningValue = getTipoContaDAO().getAllTipoContas();

        Assert.assertEquals(3, returningValue.size());
    }
}

File: app/src/androidTest/java/br/com/alma/mybudget/persistencia/persistencia/TipoContaDAOTest.kt
package br.com.alma.mybudget.persistencia.persistencia

import br.com.alma.mybudget.modelo.TipoConta
import org.junit.Assert
import org.junit.Test
import java.lang.Exception

class TipoContaDAOTest : AppDatabaseTest() {
    @Test
    @Throws(Exception::class)
    fun insertTipoConta() {
        val tipoConta = TipoConta(1, "TipoConta")

        tipoContaDAO?.insert(tipoConta)
        val returningValue: TipoConta = tipoContaDAO?.getTipoConta(tipoConta.uid)!!

        Assert.assertEquals(tipoConta.uid, returningValue.uid)
    }

    @Test
    @Throws(Exception::class)
    fun deletetipoConta() {
        val tipoConta = TipoConta(1, "TipoConta")

        tipoContaDAO?.insert(tipoConta)
        tipoContaDAO?.delete(tipoConta)
        val returningValue: TipoConta? = tipoContaDAO?.getTipoConta(tipoConta.uid)

        Assert.assertNull(returningValue)
    }


    @Test
    @Throws(Exception::class)
    fun recuperaTodosOsTipoContas() {
        var tipoConta = TipoConta(1, "Banco")
        tipoContaDAO?.insert(tipoConta)
        tipoConta = TipoConta(2, "Caixa")
        tipoContaDAO?.insert(tipoConta)
        tipoConta = TipoConta(3, "Cartão")
        tipoContaDAO?.insert(tipoConta)

        val returningValue: List<TipoConta?>? = tipoContaDAO?.getAllTipoContas()
        Assert.assertEquals(3, returningValue?.size?.toLong() ?: -1)
    }
}
